demos/demo-minimal-js/index.js

```diff
@@ -3,115 +3,154 @@
 import CallingExtensions, { Constants } from "@hubspot/calling-extensions-sdk";
 const { errorType } = Constants;
 
-const callback = () => {
-  let rowId = 0;
-  const incomingMsgContainer = document.querySelector("#incomingMsgs");
-  function appendMsg(data, event) {
-    const div = document.createElement("div");
-    rowId += 1;
-    div.innerHTML = `<span>${rowId}: </span><span>${
-      event.type
-    }, ${JSON.stringify(data)}</span>`;
-    incomingMsgContainer.append(div);
+const state = {
+  phoneNumber: "",
+  engagementId: 0,
+};
+
+function disableButtons(ids) {
+  ids.forEach(id => {
+    document.querySelector(`#${id}`).setAttribute("disabled", true);
+  });
+}
+
+function enableButtons(ids) {
+  ids.forEach(id => {
+    document.querySelector(`#${id}`).removeAttribute("disabled");
+  });
+}
+
+function initializeButtons() {
+  enableButtons(["login", "senderror", "resize"]);
+}
+
+function toggleLogin(clickedButtonId) {
+  if (clickedButtonId === "login" || clickedButtonId === "initialize") {
+    disableButtons(["login", "initialize"]);
+    enableButtons(["logout", "startcall"]);
   }
+  if (clickedButtonId === "logout") {
+    enableButtons(["login"]);
+    disableButtons(["logout", "startcall", "answercall", "endcall", "savecall"]);
+  }
+}
+
+function startCall() {
+  disableButtons(["startcall"]);
+  enableButtons(["answercall", "endcall"]);
+}
+
+function answerCall() {
+  disableButtons(["answercall"]);
+}
+
+function endCall() {
+  disableButtons(["answercall", "endcall"]);
+  enableButtons(["savecall", "startcall"]);
+}
+
+function saveCall() {
```

Comment:
Suggestion based on our discussion about having the naming match the SDK methods closely.
```suggestion
function completeCall() {
```
Could you also update the input element id in [line 97](https://github.com/HubSpot/calling-extensions-sdk/pull/113/files#diff-5c077add9a2f50f83cea34c5420e0c3758eeeec03d882f33e977101ebc44bb70R97)?

Reply:
I've changed the button value to `call completed` but kept the naming here because we actually refer to the methods as `onCallSaved` everywhere else in SDK support and HubSpot FE and even the in the React demo widget.

Reply:
@alonso-cadenas Any additional thoughts on this? I'd be down to change it to `completeCall` as well.

Reply:
> I've changed the button value to `call completed` but kept the naming here because we actually refer to the methods as `onCallSaved` everywhere else in SDK support and HubSpot FE and even in the React demo widget.

External devs don't have the context of the SDK support and HubSpot FE, that's the reason I was thinking it'd be more intuitive to match the SDK methods. I tend to think of `completed` as more of a status of the call compared to`saved` because we are also saving the call when it's started, answered, etc. Overall, I don't feel strongly enough about this to request a change. In an ideal world, whether it's saved or completed, we use the same naming across the entire stack!

Reply:
Okay this makes sense to me!